### ISSUE TITLE
[bug] <title>Failing installation of @roots/bud-typescript with typescript 5

### ISSUE THREAD

**Issue by elliason** [roots/bud#2216]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

I have a package that uses Bud and bud-typescript. I use Yarn 1 (1.22.17), Node 18 (18.15.0) and Ubuntu 22.04.2 LTS.
The problem is that I cannot install bud-typescript together with typescript@^5. Installation works with typescript@^4 bud version 5 seems to cause troubles.
The error essage: An unexpected error occurred: "could not find a copy of typescript to link in /.../theme/node_modules/@roots/bud-typescript/node_modules"
The issue is reproducible with empty package only containing typescript and bud-typescript
![image](https://user-images.githubusercontent.com/6887123/230719005-e596e34f-375e-4312-804c-474cc96eec37.png)

### Expected Behavior

It should install bud-typescript along with typescript ^5 without problems

### Actual Behavior

![image](https://user-images.githubusercontent.com/6887123/230719077-700ca091-6f13-4d73-ac23-6265744f3459.png)

### Steps To Reproduce

1. yarn init
2. yarn add typescript
3. yarn add @roots/bud-typescript -D

### version

6.12.0

**Comment by kellymears** [comment]
Can you try manually flushing the cache? 

```ts
yarn bud build --force
```

I think this is likely a cache issue with bud.js' resolver (it's still trying to resolve ts from the nested node_modules dir).

I'll look into why the cache isn't being flushed as it should be cleared out by `package.json` edits. I'm almost positive that's what's happening.

**Comment by kellymears** [comment]
It is not what I thought at all. Cache is fine. 

This is a bug in yarn v1: https://github.com/yarnpkg/yarn/issues/6285.

#2222 should sidestep it.

For now you can add typescript to your resolutions field [using the CLI](https://yarnpkg.com/cli/set/resolution) or adding it to package.json directly:

```json
{
  "resolutions": {
    "typescript": "5.0.4"
  }
}
```

**Comment by kellymears** [comment]
[This issue is addressed in 6.12.1](https://bud.js.org/releases/6.12.1#failure-installing-custom-typescript-version-with-yarn).
